crates/bevy_ecs/src/storage/table.rs

```diff
@@ -86,6 +90,19 @@ impl TableRow {
     }
 }
 
+/// A type-erased contiguous container for data of a homogenous type.
+///
+/// Conceptually, a [`Column`] is very similar to a `Vec<T>` with it's generic type erased.
+/// It also store the change detection ticks for the stored components, kept in two separate
+/// contiugous buffers internally. An element shares it's data across these buffers by using the
+/// saame index (i.e. the entity at row 3 has it's data at index 3 and it's change detection ticks at
+/// index 3). A slice to these contiguous blocks of memory can be fetched
+/// via [`Column::get_data_slice`], [`Column::get_added_ticks_slice`], and
+/// [`Column::get_changed_ticks_slice`].
+///
+/// Like many other low-level storage types, [`Column`] has a limited and highly unsafe
```

Comment:
As a side note, I find it interesting that a lot of these super useful low-level engineering tricks are currently only possible using unsafe. It makes me think to some degree performance does come at the cost of safety.

Reply:
That is indeed true. At this level, we're eschewing a lot of Rust's safety guarantees to get tight control over the performance profile of the resultant compiled code and enforcing Rust's rules ourselves.